ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java

```diff
@@ -440,7 +441,10 @@ public void updateTableOptimizeStatus() {
               getIcebergMinorPlan(fileScanTasks, -1, System.currentTimeMillis(), partitionIsRunning);
           List<BaseOptimizeTask> minorTasks = minorPlan.plan();
           // pending for minor optimize
-          if (CollectionUtils.isNotEmpty(minorTasks)) {
+          if (CollectionUtils.isNotEmpty(minorTasks) && TableOptimizeStatusUtil.in(
```

Comment:
It may be related to this issue https://github.com/NetEase/arctic/issues/959, we'd better discuss it under this issue, and then put a new PR to fix it.